### ISSUE TITLE
[Java] [Spark] Moving TransformationInfo to java client to reuse it in different integrations.

### ISSUE THREAD

**Issue by tnazarew** [OpenLineage/OpenLineage#3074]
**Details:**

Transformation type information is stored in two places, CLL facet schema and `TransformationInfo` class in spark integration. To move standardization of transformation type info handling, the class could be moved to java client so it can be used in different (potential) integrations. Is there a package in java client where it would belong?

**Comment by pawel-big-lebowski** [comment]
I think the real problem here is that `TransformationInfo` is an enum but we don't wanto to store enums in Spec as they evolve in a breaking change way. 

Storing the `TransformationInfo` class in java client is one of the workarounds for this but it is not 100% clean. 

The other one would be to extend json-schema definition for string type with possible values. The semantics here should not list all the allowed values only, but only to specify some allowed values while still allowing anything else.  This can be done with regular expressions pattern within json-schema field https://json-schema.org/understanding-json-schema/reference/string#regexp

**Comment by tnazarew** [comment]
`TransformationInfo` is more than just enum to store allowed values, it also contains the logic of merging transformations like
a -> b, b -> c into a -> c. Besides that, `TransformationInfo` objects are not in the OL event, they are just intermediary objects to make the transformation type extraction easier. So they don't really touch the spec.

**Comment by pawel-big-lebowski** [comment]
@tnazarew you're right. I mean `type` and `subtype` from `TransformationInfo` are enums.

**Comment by jphalip** [comment]
Thanks for bringing this up. To give some more context, I'm currently working on a Hive integration and `TransformationInfo` is extremely helpful, in particular because of the merge logic mentioned above. I agree that having the type and subtype in the spec would also help make things consistent.
For now, I've just copy-pasted the `TransformationInfo` class into my codebase, but it'd be great to move it to a shared location so we don't have to maintain it separately.

**Comment by jphalip** [comment]
Hi all. Just wanted to check if you had any updates on this. Thanks!
